.golangci.yml

```diff
@@ -0,0 +1,68 @@
+run:
+  timeout: 5m
+linters:
+  enable:
+    # Simple linter to check that your code does not contain non-ASCII identifiers.
+    - asciicheck
+    # Go linter that checks if package imports are in a list of acceptable packages.
+    - depguard
+    - exportloopref
+    - gofmt
+    - gosec
+    - nakedret
+    - nolintlint
+    - bidichk
+  disable:
+    - unused
+    - errcheck
+linters-settings:
+  gosec:
+    # To select a subset of rules to run.
+    # Available rules: https://github.com/securego/gosec#available-rules
+    # Default: [] - means include all rules
+    includes:
+      - G102 # Bind to all interfaces
+      - G106 # Audit the use of ssh.InsecureIgnoreHostKey
+      - G108 # Profiling endpoint automatically exposed on /debug/pprof
+      - G109 # Potential Integer overflow made by strconv.Atoi result conversion to int16/32
+      - G110 # Potential DoS vulnerability via decompression bomb
+      - G111 # Potential directory traversal
+      - G112 # Potential slowloris attack
+      - G113 # Usage of Rat.SetString in math/big with an overflow (CVE-2022-23772)
+
+  govet:
+    check-shadowing: true
+    enable-all: true
+    disable:
+      - shadow
+      - fieldalignment
+      - loopclosure
+
+  staticcheck:
+    checks: ["all", "-ST1000", "-ST1003", "-ST1016", "-ST1020", "-ST1021", "-ST1022", "-SA4006"]
+
+  bidichk:
+    # The following configurations check for all mentioned invisible unicode runes.
+    # All runes are enabled by default.
+    left-to-right-embedding: true
+    right-to-left-embedding: true
+    pop-directional-formatting: true
+    left-to-right-override: true
+    right-to-left-override: true
+    left-to-right-isolate: true
+    right-to-left-isolate: true
+    first-strong-isolate: true
+    pop-directional-isolate: true
+
+
+issues:
+  exclude-rules:
+    - linters:  
+        - staticcheck
+      text: "SA1019:"
+    - linters:
+        - gosimple
+      text: "S1009:"
```

Comment:
S1009 means that redundant nil check on slices and seems to belong to staticcheck, but why exclude it?

Reply:
> S1009 means that redundant nil check on slices and seems to belong to staticcheck, but why exclude it? Also, SA1019. we can fix it now if we use deprecated function If it's easy to modify.

You are right. they don't seem too difficult to modify. I will now attempt to make some changes and see if I can remember why I configured them that way originally